src/base/static/components/input-form/input-form.js

```diff
@@ -0,0 +1,709 @@
+            showAllValues={true}
+          />
+        ];
+        break;
+      case constants.COMMON_FORM_ELEMENT_TYPENAME:
+        const commonFormElementConfig = Object.assign(
+          {},
+          placeConfig.common_form_elements[fieldConfig.name],
+          {name: fieldConfig.name}
+        );
+        return this.buildField(commonFormElementConfig);
+        break;
+    }
+  }
+
+  validate() {
+    const { fieldValues, formValidationErrors } = this.state;
+    let newValidationErrors = [];
+
+    if (!this.geometry) {
+      newValidationErrors.push(messages.missingGeometry);
+    }
+
+    for (let i = 0; i < this.requiredFields.length; i++) {
+      if (!fieldValues[this.requiredFields[i]]) {
+        newValidationErrors.push(messages.missingRequired);
+        break;
+      }
+    }
+
+    if (newValidationErrors.length > 0) {
+      this.onInvalidSubmit(newValidationErrors);
+      return;
+    } else {
+      this.onValidSubmit();
+      return;
+    }
+  }
+
+  // due to https://stackoverflow.com/questions/8917921/cross-browser-javascript-not-jquery-scroll-to-top-animation
+  scrollTo(elt, to, duration) {
+    const difference = to - elt.scrollTop;
+    const perTick = difference / duration;
+    setTimeout(() => {
+      elt.scrollTop = elt.scrollTop + perTick;
+      if (elt.scrollTop === to) return;
+      this.scrollTo(elt, to, duration - 10);
+    }, 10);
+  }
+
+  onInvalidSubmit(errors) {
+    this.setState({ formValidationErrors: errors });
+    this.scrollTo(document.querySelector(this.props.container), 0, 300);
+  }
+
+  onValidSubmit() {
+    
+    // TODO: this state should disable individual fields as well (?), not just
+    //       the submit button.
+    this.setState({ formIsSubmitting: true });
+
+    const { collectionsSet, placeConfig } = this.props;
+    const { coverImages, fieldValues, selectedCategory } = this.state;
+    const spinnerTarget = document.getElementsByClassName("input-form__submit-spinner")[0];
+    const selectedCategoryConfig = placeConfig.place_detail
+            .find(category => category.category === selectedCategory);
+    let collection = collectionsSet.places[selectedCategoryConfig.dataset];
+    let model;
+    let attrs;
+
+    new Spinner(Shareabouts.smallSpinnerOptions).spin(spinnerTarget);
+
+    collection.add({
+      location_type: selectedCategoryConfig.category,
+      datasetSlug: this.props.mapConfig.layers.find(
+        layer => selectedCategoryConfig.dataset === layer.id
+      ).slug,
+      datasetId: selectedCategoryConfig.dataset,
+      showMetadata: selectedCategoryConfig.showMetadata,
+    });
+    model = collection.at(collection.length - 1);
+    attrs = Object.assign({}, fieldValues);
+
+    this.richTextFields.forEach(fieldName => {
+      // replace base64 image data with placeholders built from each image's name
+      attrs[fieldName] = attrs[fieldName].replace(
+        /\<img.*?name=\"(.*?)\".*?\>/g,
+        "{{#rich_text_image $1}}"
```

Comment:
@modulitos -- here's the markup I decided to use for rich text-embedded images. So rich text with images will look something like this:
```
<p>some rich text</p>{{#rich_text_image sf35szg}}<span>some more rich text</span>
```
where `sf35szg` corresponds to the name of an attachment model associated with the given place.

Do you think that's a good way to mark up rich text images? It's sort of inspired by Handlebars markup.